src/components/message/message.tsx

```diff
@@ -0,0 +1,199 @@
+ *
+ * There are 5 different types of messages available to user. You can pass the string which you want to render in `Message` component as argument.
+ *
+ * * `message.info()`
+ * * `message.success()`
+ * * `message.error()`
+ * * `message.warning()`
+ * * `message.loading()`
+ *
+ * Use `removeMessage()` method to programmatically remove a message. `removeMessage()` takes the id of `Message` component you want to remove.
+ * All `message` methods return the id of the message which you can use with `removeMessage()`
+ */
+
+export function MessageProvider({
+  children,
+  defaultDismissTime = 3000,
+}: MessageProviderProps) {
+  const toastContainer = useRef<HTMLDivElement>(document.createElement('div'))
+
+  useLayoutEffect(() => {
+    const container = toastContainer.current
+    document.body.appendChild(container)
+    return () => {
+      document.body.removeChild(container)
+    }
+  }, [])
+
+  const [messages, setMessages] = useState<MessageListType[]>([])
+
+  const removeMessage = (id: string) => {
+    setMessages((prevState) => {
+      if (prevState.find((message) => message.id === id)) {
+        return prevState.filter((message) => message.id !== id)
+      }
+      return prevState
+    })
+  }
+
+  const dispatchMessage = (type: MessageTypes) => (
+    title: string,
+    dismissTime?: number,
+  ) => {
+    const messageId = (
+      Math.random().toString(36) + Date.now().toString(36)
+    ).substr(2, 10)
+
+    setMessages((prevState) => [...prevState, { id: messageId, title, type }])
+
+    setTimeout(() => {
+      removeMessage(messageId)
+    }, dismissTime ?? defaultDismissTime)
+
+    return messageId
+  }
+
+  const info = dispatchMessage(MessageTypes.INFO)
+  const success = dispatchMessage(MessageTypes.SUCCESS)
+  const warning = dispatchMessage(MessageTypes.WARNING)
+  const error = dispatchMessage(MessageTypes.ERROR)
+  const loading = dispatchMessage(MessageTypes.LOADING)
+
+  return (
+    <>
+      <MessageContext.Provider
+        value={{
+          message: { info, success, warning, error, loading },
+          removeMessage,
+        }}
+      >
+        {children}
+      </MessageContext.Provider>
+      {createPortal(
+        <TransitionGroup className="fixed inset-0 z-10 flex flex-col items-center justify-start px-4 py-6 space-y-4 pointer-events-none">
+          {messages.map((message) => (
+            <CSSTransition
+              timeout={200}
+              classNames="toast-right"
+              key={message.id}
+              unmountOnExit
+            >
+              <Message message={message} />
+            </CSSTransition>
+          ))}
+        </TransitionGroup>,
+        toastContainer.current,
+      )}
+    </>
+  )
+}
+
+type MessageProps = {
+  message: MessageListType
+  closeIcon?: boolean
+}
+
+function Message({ message }: MessageProps) {
+  const toastIcon = useCallback((type) => {
```

Comment:
We don't need to use `useCallback`, instead we can use `useMemo` as the `toastIcon` should be a `React.Element`. For this we don't need to create a function and call that function to get that element.